packages/grafana-llm-frontend/src/mcp.tsx

```diff
@@ -206,6 +240,10 @@ function createClientResource(appName: string, appVersion: string): ClientResour
   return {
     read() {
       if (status === 'pending') {
+        if (!result?.enabled) {
+          console.warn('MCP is not enabled');
+          return result || { client: null, enabled: false };
+        }
         throw promise;
```

Comment:
@sd2k tested locally the changes, and found out that our problem is here. I added this logic just to markup, but the fact is that when the client returns 404 (e.g. MCP not enabled), it enters in an infinite state of `pending`. The problem with this `!result?.enabled` check is that this is async, so not reliable as its initial state will be always `null | false`.